### ISSUE TITLE
[Points] Unable to change symbol of points

### ISSUE THREAD

**Issue by psobolewskiPhD** [napari/napari#7139]
### 🐛 Bug Report

If one creates a points layer and makes some points, it appears impossible to change the symbol type afterwards.
Changing the symbol with a point selected returns:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
File ~/Dev/napari/napari/_qt/layer_controls/qt_points_controls.py:218, in QtPointsControls.changeCurrentSymbol(self=<napari._qt.layer_controls.qt_points_controls.QtPointsControls object>, text='diamond')
    210 """Change marker symbol of the points on the layer model.
    211 
    212 Parameters
   (...)
    215     Index of current marker symbol of points, eg: '+', '.', etc.
    216 """
    217 with self.layer.events.symbol.blocker(self._on_current_symbol_change):
--> 218     self.layer.current_symbol = SYMBOL_TRANSLATION_INVERTED[text]
        SYMBOL_TRANSLATION_INVERTED[text] = <Symbol.DIAMOND: 'diamond'>
        self.layer = <Points layer 'Points' at 0x130acb110>
        SYMBOL_TRANSLATION_INVERTED = {'arrow': <Symbol.ARROW: 'arrow'>, 'clobber': <Symbol.CLOBBER: 'clobber'>, 'cross': <Symbol.CROSS: 'cross'>, 'diamond': <Symbol.DIAMOND: 'diamond'>, 'disc': <Symbol.DISC: 'disc'>, 'hbar': <Symbol.HBAR: 'hbar'>, 'ring': <Symbol.RING: 'ring'>, 'square': <Symbol.SQUARE: 'square'>, 'star': <Symbol.STAR: 'star'>, 'tailed arrow': <Symbol.TAILED_ARROW: 'tailed_arrow'>, 'triangle down': <Symbol.TRIANGLE_DOWN: 'triangle_down'>, 'triangle up': <Symbol.TRIANGLE_UP: 'triangle_up'>, 'vbar': <Symbol.VBAR: 'vbar'>, 'x': <Symbol.X: 'x'>}
        text = 'diamond'
        self = <napari._qt.layer_controls.qt_points_controls.QtPointsControls object at 0x130bdad50>

File ~/Dev/napari/napari/layers/points/points.py:977, in Points.current_symbol(self=<Points layer 'Points'>, symbol=<Symbol.DIAMOND: 'diamond'>)
    975 self._current_symbol = symbol
    976 if self._update_properties and len(self.selected_data) > 0:
--> 977     self.symbol[list(self.selected_data)] = symbol
        symbol = <Symbol.DIAMOND: 'diamond'>
        self = <Points layer 'Points' at 0x130acb110>
    978     self.events.symbol()
    979 self.events.current_symbol()

ValueError: assignment destination is read-only
```
Deselecting all points and changing the symbol in the dropdown works, but then newly placed points still use the previous symbol.

If one makes a fresh points layer and changes the symbol -- **prior to placing any points** -- then the alternate symbol is used, but again can't be changed thereafter as noted above.

### 💡 Steps to Reproduce

1. create a Points layer
2. add points using the GUI
3. Try to change the symbol using the dropdown to a different symbol (e.g. disc to square)

### 💡 Expected Behavior

The tooltip for symbol says it will change the symbol of the currently selected points and those placed thereafter, so I would expect this to occur.

### 🌎 Environment

napari: 0.5.1rc1.dev1+g5535c71a7
Platform: macOS-14.5-arm64-arm-64bit
System: MacOS 14.5
Python: 3.12.4 | packaged by conda-forge | (main, Jun 17 2024, 10:13:44) [Clang 16.0.6 ]
Qt: 6.7.1
PyQt6: 
NumPy: 2.0.0
SciPy: 1.14.0
Dask: 2024.7.1
VisPy: 0.14.3
magicgui: 0.8.3
superqt: 0.6.7
in-n-out: 0.2.1
app-model: 0.2.8
npe2: 0.7.6

OpenGL:
- GL version: 2.1 Metal - 88.1
- MAX_TEXTURE_SIZE: 16384
- GL_MAX_3D_TEXTURE_SIZE: 2048

Screens:
- screen 1: resolution 1680x1050, scale 2.0

Optional:
- numba: 0.60.0
- triangle not installed
- napari-plugin-manager: 0.1.0a3.dev189+gc48deb2

Settings path:
- /Users/piotrsobolewski/Library/Application Support/napari/napari-dev_a1eb8b76ba95fa16ad06e26097b46b8455dfbf0b/settings.yaml

### 💡 Additional Context

I checked and the behavior is as expected in 0.4.19post1

**Comment by psobolewskiPhD** [comment]
Bisect points to 
0e6f0af3019db6d67fabeb3e051a76244bc0aeaf
cc: @Czaki

**Comment by psobolewskiPhD** [comment]
Checking `viewer.layers[0].symbol.flags.writeable` it returns `False` and doing:
`viewer.layers[0].symbol.flags.writeable = True` returns the expected behavior.

**Comment by psobolewskiPhD** [comment]
OK, I think this line:
https://github.com/napari/napari/blob/5535c71a787a5a17dc1ab8147e23b3bd0eff3131/napari/layers/points/points.py#L962
Is returning a read-only view, based on https://numpy.org/doc/stable/reference/generated/numpy.broadcast_to.html#numpy-broadcast-to

Seems like we need to make a copy so that it can be edited safely.

**Comment by jni** [comment]
Closed by #7184
